mlir/lib/Conversion/FuncToLLVM/FuncToLLVM.cpp

```diff
@@ -62,6 +62,7 @@ using namespace mlir;
 static constexpr StringRef varargsAttrName = "func.varargs";
 static constexpr StringRef linkageAttrName = "llvm.linkage";
 static constexpr StringRef barePtrAttrName = "llvm.bareptr";
+static constexpr StringRef noInlineAttrName = "no_inline";
```

Comment:
I'd rather not rely on magic constants in local files and get this somehow from the interface or from specific op classes.

Reply:
Thank you for the comment, @ftynse!

Would that be acceptable to add the get/set interface methods into `FunctionOpInterface` and then implement it for `func.func`, `llvm.func`, etc.?  Or did you mean to suggest something else?

Reply:
I didn't have a specific suggestion, just observed that such action-at-a-distance is dangerous. The op definition may change the name of the attribute to, e.g., `noinline` without the underscore, and this code will silently break. Adding interface methods would make sense to be, doubly so if these methods are then used in the inliner logic.

Reply:
> doubly so if these methods are then used in the inliner logic.

Thanks! It sounds quite reasonable to add such interface methods in `FunctionOpInterface`.